From 77e0afea18fda30d0831736deae79bfe82d567f9 Mon Sep 17 00:00:00 2001
From: Valentin Churavy <v.churavy@gmail.com>
Date: Tue, 7 Feb 2023 16:42:43 -0500
Subject: [PATCH] Don't build static lib

---
 CMakeLists.txt | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb0a30a..f5aef45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -342,47 +342,47 @@ set_target_properties(amgx_libs PROPERTIES CUDA_ARCHITECTURES "${CUDA_ARCH}")
 target_compile_options(amgx_libs PUBLIC $<$<COMPILE_LANGUAGE:CUDA>: ${CUDA_NVCC_FLAGS} >)
 
 # build amgx
-add_library(amgx STATIC $<TARGET_OBJECTS:amgx_libs>)  # static lib
-target_link_libraries(amgx amgx_libs)
+#add_library(amgx STATIC $<TARGET_OBJECTS:amgx_libs>)  # static lib
+#target_link_libraries(amgx amgx_libs)
 
 add_library(amgxsh SHARED $<TARGET_OBJECTS:amgx_libs>)  # shared lib
 target_link_libraries(amgxsh amgx_libs)
 
-set_target_properties(amgx PROPERTIES LINKER_LANGUAGE CUDA)
+#set_target_properties(amgx PROPERTIES LINKER_LANGUAGE CUDA)
 set_target_properties(amgxsh PROPERTIES LINKER_LANGUAGE CUDA)
 
-target_compile_options(amgx PUBLIC $<$<COMPILE_LANGUAGE:CUDA>: ${CUDA_NVCC_FLAGS} >)
+#target_compile_options(amgx PUBLIC $<$<COMPILE_LANGUAGE:CUDA>: ${CUDA_NVCC_FLAGS} >)
 target_compile_options(amgxsh PUBLIC $<$<COMPILE_LANGUAGE:CUDA>: ${CUDA_NVCC_FLAGS} >)
 
 IF (WIN32)
-  target_link_libraries( amgx CUDA::cublas CUDA::cusparse CUDA::cusolver)
+	#target_link_libraries( amgx CUDA::cublas CUDA::cusparse CUDA::cusolver)
   target_link_libraries( amgxsh CUDA::cublas CUDA::cusparse CUDA::cusolver)
 ELSE (WIN32)
-  target_link_libraries( amgx CUDA::cublas CUDA::cusparse CUDA::cusolver m pthread)
+	#target_link_libraries( amgx CUDA::cublas CUDA::cusparse CUDA::cusolver m pthread)
   target_link_libraries( amgxsh CUDA::cublas CUDA::cusparse CUDA::cusolver  m pthread)
 ENDIF(WIN32)
 
 if(MPI_FOUND)
-    target_link_libraries( amgx   ${MPI_C_LIBRARIES})
+	#target_link_libraries( amgx   ${MPI_C_LIBRARIES})
     target_link_libraries( amgxsh ${MPI_C_LIBRARIES})
 endif(MPI_FOUND)
 
 #link magma
 if (MAGMA_ROOT_DIR)
-  target_link_libraries(amgx   ${magma_libs})
+	#target_link_libraries(amgx   ${magma_libs})
   target_link_libraries(amgxsh ${magma_libs})
 endif(MAGMA_ROOT_DIR)
 #link lapack (MKL)
 if (MKL_ROOT_DIR)
-  target_link_libraries(amgx   ${mkl_libs} )
+	#target_link_libraries(amgx   ${mkl_libs} )
   target_link_libraries(amgxsh ${mkl_libs})
 endif(MKL_ROOT_DIR)
 
 # set arch for main libs
-set_target_properties(amgx PROPERTIES CUDA_ARCHITECTURES "${CUDA_ARCH}")
+#set_target_properties(amgx PROPERTIES CUDA_ARCHITECTURES "${CUDA_ARCH}")
 set_target_properties(amgxsh PROPERTIES CUDA_ARCHITECTURES "${CUDA_ARCH}")
 
-install(TARGETS amgx   DESTINATION "lib")
+#install(TARGETS amgx   DESTINATION "lib")
 install(TARGETS amgxsh DESTINATION "lib")
 
 #export(TARGETS amgxsh FILE ${CMAKE_SOURCE_DIR}/amgxsh.cmake)
-- 
2.37.2

